llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -1753,8 +1756,7 @@ static Value *generateNewInstTree(ArrayRef<InstLane> Item, FixedVectorType *Ty,
 // do so.
 bool VectorCombine::foldShuffleToIdentity(Instruction &I) {
   auto *Ty = dyn_cast<FixedVectorType>(I.getType());
-  if (!Ty || !isa<Instruction>(I.getOperand(0)) ||
-      !isa<Instruction>(I.getOperand(1)))
```

Comment:
Isn't this an unrelated change?

Reply:
Sure - I can split it out. It will be needed for many of the tests, but there should be enough to test various cases still.